### PR TITLE
Added ability to specify default headers in mail client

### LINKED ISSUE #22: Default headers
It would be nice to be able to set default headers which will be sent with every message. Like the `DefaultFrom`.

### PR DESCRIPTION
**Pull request by ricardoboss**
Also made MailHeaderCollection keys case insensitive.

Fixes #22

### CODE REVIEW

File: TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs [hunk 1]
```diff
@@ -158,7 +158,31 @@ private IEnumerable<MailAddress> GetAddresses(string key)
...
     public IReadOnlyDictionary<string, object> NonAddressHeaders => this
         .ExceptBy(AddressKeys, x => x.Key)
         .ToDictionary(x => x.Key, x => x.Value);
+
+    /// <summary>
+    /// Appends <paramref name="other"/> to this collection without overwriting existing values.
+    /// </summary>
+    /// <param name="other">The collection to append.</param>
+    /// <returns>The current instance.</returns>
+    public MailHeaderCollection Append(MailHeaderCollection other)
```
Comment (chucker) [inline TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs:173]: I wonder if `bool overwrite = false` would be useful, in the spirit of letting others use this as well.

File: TRENZ.Lib.RazorMail.Tests/MailHeaderCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+{
+    [Test]
+    public void TestAppendDoesntOverwriteExistingValues()
+    {
+        var collection = new MailHeaderCollection
+        {
+            { "key", "value" },
+        };
+
+        collection.Append(new()
```
Comment (chucker) [inline TRENZ.Lib.RazorMail.Tests/MailHeaderCollectionTests.cs:17]: it's only a test, but I would prefer an explicit type here

**Review by chucker — CHANGES_REQUESTED** [review]

File: TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs (hunk 1 above)
Comment (ricardoboss) [reply TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs]: I added a separate method instead

File: TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs [hunk 3]
```diff
@@ -170,7 +170,7 @@ private IEnumerable<MailAddress> GetAddresses(string key)
     /// </summary>
     /// <param name="other">The collection to append.</param>
     /// <returns>The current instance.</returns>
-    public MailHeaderCollection Append(MailHeaderCollection other)
+    public MailHeaderCollection AppendFrom(MailHeaderCollection other)
```
Comment (chucker) [inline TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs:173]: Hm, `From` seems like poor naming, given that it also has a more specific meaning (see, for example, `DefaultFrom`).

**Review by chucker — CHANGES_REQUESTED** [review]

**Comment by ricardoboss** [comment]
@chucker can you re-review?

**Review by chucker — APPROVED** [review]

**Pull request merged**